File: 1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one.java
class Solution {
    public int numSteps(String s) {
        int last = s.length() - 1, carry = 0, result = 0;

        while (last > 0) {
            int chk = s.charAt(last) - '0' + carry;
            if (chk == 0) result++;
            else if (chk == 2) result++;
            else {
                carry = 1;
                result += 2;
            }
            last--;
        }

        if (carry == 1) result++;

        return result;
    }
}

File: 1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one.py
class Solution:
    def numSteps(self, s: str) -> int:
        target = 0
        result = 0

        for i in range(len(s) - 1, -1, -1):
            if s[i] == '1':
                target += 2 ** (len(s) - 1 - i)

        while target != 1:
            result += 1
            if target % 2:
                target += 1
            else:
                target = target >> 1

        return result